Lesson15/duygu/script.js

```diff
@@ -0,0 +1,631 @@
+     - Logs "Odd" if the number is odd
+*/
+function evenOrOdd(number) {
+  if (number % 2 === 0) {
+    console.log("Even");
+  } else {
+    console.log("Odd")
+  }
+}
+
+evenOrOdd(2);
+evenOrOdd(7);
+
+
+/*
+11. URL Protocol Checker
+   - Define a function `checkProtocol(url)` that converts the URL to lowercase
+     and checks if it starts with "https" using .startsWith().
+   - Log "Secure connection" if true, otherwise "Unsecure connection".
+*/
+
+function checkProtocol(url) {
+  const securedCon = url.toLowerCase().startsWith("https");
+  if (securedCon ) {
+    console.log("Secure connection" );
+  } else {
+    console.log("Unsecure connection");
+  }
+}
+checkProtocol("HTTP://WWW.HELLOWORLD.COM");
+checkProtocol("https://www.helloworld.com");
+
+
+
+/*
+12. Switch: Day of the Week
+   - Define a function `getDayOfWeek(num)` that uses a switch statement:
+     1 -> "Monday"
+     2 -> "Tuesday"
+     ...
+     7 -> "Sunday"
+     - Log the matched day or "Invalid day" if out of range.
+*/
+
+function getDayOfWeek(num) {
+  switch (num) {
+    case 1:
+      console.log("Monday");
+      break;
+    case 2:
+      console.log("Tuesday");
+      break;
+    case 3:
+      console.log("Weddnesday");
+      break;
+    case 4:
+      console.log("Thursday");
+      break;
+    case 5:
+      console.log("Friday");
+      break;
+    case 6:
+      console.log("Saturday");
+      break;
+    case 7:
+      console.log("Sunday");
+      break;
+    default:
+      console.log("Unknow day");
+      break;
+  }
+}
+
+getDayOfWeek(5);
+getDayOfWeek(4);
+getDayOfWeek(1);
+getDayOfWeek('hello');
+getDayOfWeek(10);
+
+
+
+/*
+13. Repeat a String
+   - Define a function `repeatWord(word, times)` that uses the .repeat() method
+     to repeat `word` `times` times.
+   - Log the repeated result.
+*/
+function repeatWord(word, times) {
+  const repeated = word.repeat(times);
+  console.log(repeated);
+}
+
+repeatWord("Hohoo", 4);
+
+/*
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+function censorWord(sentence, censored){
+  console.log (sentence.replaceAll(censored, '****'));
+}
+censorWord("This is a censored word.", "censored");
+
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/
+function startsWithA(str) {
+  if (str.charAt(0) === 'A') {
+  console.log("Starts with A");
+}
+ else {
+  console.log("Does not start with A");
+}
+}
+startsWithA("Ayakkabı");
+startsWithA("Pantolon");
+/*
+16. Slice Last N Characters
+   - Define a function `sliceLastN(text, n)` that uses .slice(-n) to extract
+     the last `n` characters of `text`.
+   - Log the result.
+*/
+function sliceLastN(text, n) {
+  const nthword = text.slice(n)
```

Comment:
In this function, the goal was to pass a positive number when calling it but have the '-' inside the function.
```suggestion
  const nthWord = text.slice(-n);
```

Having to provide a positive number usually makes it easier to work with a function.